### DISCUSSION TITLE
how to remove null values when using this fillter sample,

### DISCUSSION THREAD

**Discussion by shanmugamani96** [PHPOffice/PhpSpreadsheet#2736] [Q&A] [answered]
[PhpSpreadsheet](https://github.com/PHPOffice/PhpSpreadsheet)/[samples](https://github.com/PHPOffice/PhpSpreadsheet/tree/master/samples)/[Reader](https://github.com/PHPOffice/PhpSpreadsheet/tree/master/samples/Reader)/11_Reading_a_workbook_in_chunks_using_a_configurable_read_filter_(version_1).php /,

using this file ,i try fillter rows and columns and convert into html ,but when i limit 1,10 rows it works good,but if i limit 3,5,then 1 and 2 rows produce null values,like attached file
![error](https://user-images.githubusercontent.com/97212148/162555230-834828e4-f08e-4610-8d2b-c8e16a5aa4cc.png)

how do remove null values?

**Comment by MarkBaker** [comment]
The filter is using the `toArray()` method to create an array to display; and `toArray()` always starts from cell A1... As the chunked reader hasn't read any data for cells in row 1 or 2, it still has to display something for those rows, so it uses a row of nulls.

If you wanted to create an array with only the rows that had data read by the chunk filter, then you would need to use `rangeToArray()` with a range argument for the cells that had been loaded from the file.(e.g. `rangeToArray('A3:D4')`.
Similarly, if you're using the HTML Writer, it will still start writing from cell A1, and will create empty rows where there is no data to display; but there is no mechanism built-in that tells the HTML Writer not to write specific rows. If this is something that you need, then I suggest that you raise a new Issue as a feature request.

**Comment by MarkBaker** [comment, marked as answer]
If you want to eliminate those empty lines, you could possibly use a callback with `setEditHtmlCallback()`

**Reply by shanmugamani96** [reply]
ok , i will check it,
